src/perfoutputwidgetkonsole.cpp

```diff
@@ -0,0 +1,166 @@
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "perfoutputwidgetkonsole.h"
+
+#include <QEvent>
+#include <QKeyEvent>
+#include <QStandardPaths>
+#include <QTemporaryFile>
+#include <QVBoxLayout>
+
+#include <KParts/ReadOnlyPart>
+#include <KParts/kde_terminal_interface.h>
+#include <KService>
+
+QString PerfOutputWidgetKonsole::m_tailPath;
+
+PerfOutputWidgetKonsole::PerfOutputWidgetKonsole(KParts::ReadOnlyPart* part, QWidget* parent)
+    : PerfOutputWidget(parent)
+    , m_konsolePart(part)
+{
+    auto layout = new QVBoxLayout(this);
+    setLayout(layout);
+
+    addPartToLayout();
+}
+
+PerfOutputWidgetKonsole::~PerfOutputWidgetKonsole() = default;
+
+PerfOutputWidgetKonsole* PerfOutputWidgetKonsole::create(QWidget* parent)
+{
+    m_tailPath = findTail();
+    if (m_tailPath.isEmpty()) {
+        return nullptr;
+    }
+
+    auto part = createPart();
+    if (!part) {
+        return nullptr;
+    }
+
+    const auto terminalInterface = qobject_cast<TerminalInterface*>(part);
+    if (!terminalInterface) {
+        qWarning("konsole kpart doesn't implement terminal interface");
+        delete part;
+        return nullptr;
+    }
+
+    return new PerfOutputWidgetKonsole(part, parent);
+}
+
+bool PerfOutputWidgetKonsole::eventFilter(QObject* watched, QEvent* event)
+{
+    Q_UNUSED(watched);
+    if (event->type() == QEvent::KeyPress) {
+        auto keyEvent = static_cast<QKeyEvent*>(event);
+
+        // ignore ctrl + c
+        if (keyEvent->key() == Qt::Key_C && keyEvent->modifiers() == Qt::ControlModifier) {
+            return true;
+        } else if (keyEvent->key() == Qt::Key_S && keyEvent->modifiers() == Qt::ControlModifier) {
+            // ignore ctrl + s (stop output)
+            return true;
+        }
+
+        if (!m_inputEnabled) {
+            // eat all key events if input is disabled
+            return true;
+        }
+    } else if (event->type() == QEvent::MouseButtonPress) {
+        auto mouseEvent = static_cast<QMouseEvent*>(event);
+
+        // prevent the user from opening the context menu
+        if (mouseEvent->button() == Qt::MouseButton::RightButton) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void PerfOutputWidgetKonsole::addOutput(const QString& output)
+{
+    m_konsoleFile->write(output.toUtf8());
+    m_konsoleFile->flush();
+}
+
+void PerfOutputWidgetKonsole::clear()
+{
+    if (m_konsolePart) {
+        m_konsolePart->deleteLater();
+    }
+
+    m_konsolePart = createPart();
+
+    if (m_konsoleFile) {
+        m_konsoleFile->deleteLater();
+    }
+
+    addPartToLayout();
+}
+
+void PerfOutputWidgetKonsole::enableInput(bool enable)
+{
+    if (enable != m_inputEnabled) {
+        m_inputEnabled = enable;
+    }
+}
+
+void PerfOutputWidgetKonsole::setInputVisible(bool visible)
+{
+    Q_UNUSED(visible);
+}
+
+QString PerfOutputWidgetKonsole::findTail()
```

Comment:
this should be a plain free function in an anon namespace in this `.cpp` file - `private static` is a code smell to me, similar to `private slots`